### DISCUSSION TITLE
Why does rio-tiler log Different ThreadPool worker counts?

### DISCUSSION THREAD

**Discussion by drnextgis** [stac-utils/titiler-pgstac#182] [General]
I am using `titiler-pgstac` with `MOSAIC_CONCURRENCY=20` and have enabled `DEBUG` logging for `rio-tiler`. When I request a tile, I notice something in the logs that I can't explain:
```
eoapi.raster  | [2024-08-23 11:47:18,124 +0000] [6] [DEBUG] rio-tiler: Running tasks in ThreadPool with max_workers=20
eoapi.raster  | [2024-08-23 11:47:18,129 +0000] [6] [DEBUG] rio-tiler: Running tasks in ThreadPool with max_workers=40
eoapi.raster  | [2024-08-23 11:47:18,135 +0000] [6] [DEBUG] rio-tiler: Running tasks in ThreadPool with max_workers=40
```

Are two different ThreadPools being used here? And why does one of them have twice the number of workers?

**Comment by drnextgis** [comment]
I believe I understand:
* `MOSAIC_CONCURRENCY` controls the process of obtaining data from a remote source.
* `RIO_TILER_MAX_THREADS` manages the concurrent reading of each raster, with a default value of 5 times the CPU count.

**Reply by vincentsarago** [reply]
Yeah there are two levels of `thread` workers 
- mosaic items 
- assets reading (because each items can have multiple assets)
